sdk/trace-shaded-deps/src/main/java/io/opentelemetry/sdk/trace/internal/JcTools.java

```diff
@@ -42,5 +43,29 @@ public static long capacity(Queue<?> queue) {
     }
   }
 
+  /**
+   * Remove up to <i>maxExportBatchSize</i> elements from the {@link Queue} and hand to consume.
+   *
+   * @throws IllegalArgumentException consumer is {@code null}
+   * @throws IllegalArgumentException if maxExportBatchSize is negative
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> void drain(Queue<T> queue, int maxExportBatchSize, Consumer<T> consumer) {
```

Comment:
This is a general purpose method so lets rename `maxExportBatchSize` to `limit` to match the naming from `MessagePassingQueue#drain`. Same applies to `drainNonJcQueue`.